core/src/saros/activities/FileActivity.java

```diff
@@ -36,6 +36,8 @@
 
   @XStreamAsAttribute protected final String encoding;
 
+  @XStreamAsAttribute protected final String fileType;
```

Comment:
Please document that this is only used in IntelliJ and maybe link to #683.

Reply:
@prechelt suggested that this could be replaced with an `isBinary()` flag as this piece of information can be provided IDE-independently. This flag would then be used to open the file as a normal text file on the other side if it is not a binary file, allowing for a base document type to work with. If the user then want's a specific syntax highlighting, they could still change the file type association by hand.

The big improvement of the approach is that it would be supported across IDEs, meaning it would not cause any issues in cross-IDE sessions.

This sounds like a good compromise between providing the best user experience and working towards cross-IDE-compatibility to me. I will check whether this approach is doable and adjust the PR accordingly.

I can still add a note that it is currently only used in IntelliJ with the new logic, but I am not sure that it is still needed in that case as the provided information is IDE-independent.

Reply:
Ok, this is not as easy as I first thought. Apparently Eclipse has a more relaxed approach to handling unknown file types. If it does not know the extension (or there is no extension), it just opens it as a plain text document. And from having a brief look at their `IResource` and `IFile` API, I couldn't find a method to determine the file type besides `getFileExtension()`, which simply returns the file extension if it exists.

So it seems like we have to guess whether a resource is binary or not (i.e. use other libraries to determine it). This also leads to other compatibility issues down the line where Eclipse simply opens an editor for a file where IntelliJ won't as it sees it as a binary file.

Reply:
Eclipse do not has a relaxed approach. You register document providers, telling Eclipse that you can handle file extension XYZ. If there is no such document provider the default one is used which simply loads the content of the (binary) file. So the Eclipse File API is not relevant for the Editor stuff.

Reply:
> 
> 
> Please document that this is only used in IntelliJ and maybe link to #683.

I do not think that it is very helpful to add specific IDE documentation into the core. In general this is an optional hint, i.e the IntelliJ part should not assume that it contains any content when regarding cross IDE-support.

Reply:
@srossbach Do you know whether there is an Eclipse API for determining whether an `IFile` is a binary file or not?

Reply:
I do not think so. This is the by far the lowest implementation you can access https://help.eclipse.org/kepler/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fcore%2Ffilesystem%2FEFS.html and it seems to only support Unix like attributes for files.

Reply:
Only found this on Stackoverflow https://stackoverflow.com/questions/36018866/how-to-determine-if-a-file-is-a-text-file-in-eclipse

But the Eclipse documentation regarding the classes is not very helpful.